w3af/plugins/crawl/open_api.py

```diff
@@ -407,6 +415,13 @@ def get_options(self):
         o = opt_factory('discover_fuzzable_headers', self._discover_fuzzable_headers, d, BOOL, help=h)
         ol.add(o)
 
+        d = 'Path to a file with parameter values.'
+        h = ('This option sets a path to a YAML file which contains parameter values'
+             ' which should be used in testing API endpoints. If no parameter values are provided,'
+             ' the plugin tries to guess them.')
+        o = opt_factory('parameter_values_location', self._parameter_values_location, d, INPUT_FILE, help=h)
```

Comment:
Maybe the YAML format validation should be done here? That could be done by changing the `INPUT_FILE` type to `YAML_INPUT_FILE` and creating the logic for handling that.

It should be fairly easy, take a look at:
 * https://github.com/andresriancho/w3af/blob/master/w3af/core/data/options/input_file_option.py
 * https://github.com/andresriancho/w3af/blob/356b14b975039706f4fd7f4f5db5b114cd75f14e/w3af/core/data/options/tests/test_input_file_option.py
 * https://github.com/andresriancho/w3af/blob/356b14b975039706f4fd7f4f5db5b114cd75f14e/w3af/core/data/options/opt_factory.py#L43

Reply:
Hmm, that may be nice. Although `input_file_option.py` doesn't look that simple at first glance :) Okay, I'll see what I can do here.

Reply:
Yeah, `input_file_option.py` doesn't look simple, but `yaml_file_option.py` should be simple :-)

Take a look at the other files in the same directory, those will give you ideas on how simple the inputs can actually be.